internal/throttler/throttler_test.go

```diff
@@ -0,0 +1,42 @@
+package throttler
+
+import (
+	"context"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+func mockThrottlerTest(ctx context.Context, throttler Throttler, r *int) {
+	throttler.Throttle(ctx)
+	*r++
+}
+
+func TestDispatchThrottler(t *testing.T) {
+	t.Run("throttler_will_release_only_when_ticked", func(t *testing.T) {
```

Comment:
this test panics if you run it more than once

```go
-dispatch-throttling) $ go test . -count=2 -race -run "TestDispatchThrottler"
--- FAIL: TestDispatchThrottler (0.00s)
    --- FAIL: TestDispatchThrottler/throttler_will_release_only_when_ticked (0.00s)
panic: duplicate metrics collector registration attempted [recovered]
        panic: duplicate metrics collector registration attempted

goroutine 41 [running]:
testing.tRunner.func1.2({0x10521dfa0, 0xc00011c6e0})
        /Users/maria.inesparnisari/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.21.9.darwin-arm64/src/testing/testing.go:1545 +0x2c8
testing.tRunner.func1()
```

you need to either

1. change `Close()` to deregister the metrics 

```go
func (r *throttler) Close() {
	prometheus.Unregister(r.throttlingDelayMsHistogram)
	r.done <- struct{}{}
}
```

and document that you have to call Close after you're done, or

2. (my preference) don't register the metrics in the constructor, do it outside. I see that you do that because you want to customize the metric name. Is that really necessary? You will already know which endpoint got throttled based on the `grpc_method` label.

Reply:
Going with option 2. We are still having a different label called "throttler_name" as the same endpoint like Check can have 2 different thorttler s for dispatch and database throttling